commands/azureCommands/create-registry.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { Registry, RegistryNameStatus } from "azure-arm-containerregistry/lib/models";
+import { SubscriptionModels } from 'azure-arm-resource';
+import { ResourceGroup } from "azure-arm-resource/lib/resource/models";
+import * as vscode from "vscode";
+import { dockerExplorerProvider } from '../../dockerExtension';
+import { ext } from '../../extensionVariables';
+import { AzureUtilityManager } from '../../utils/azureUtilityManager';
+import { quickPickLocation, quickPickResourceGroup, quickPickSKU, quickPickSubscription } from '../utils/quick-pick-azure';
+
+/* Creates a new Azure container registry based on user input/selection of features */
+export async function createRegistry(): Promise<Registry> {
+    const subscription: SubscriptionModels.Subscription = await quickPickSubscription();
+    const resourceGroup: ResourceGroup = await quickPickResourceGroup(true, subscription);
+    const client = AzureUtilityManager.getInstance().getContainerRegistryManagementClient(subscription);
+    const registryName: string = await acquireRegistryName(client);
+    const sku: string = await quickPickSKU();
+    const location = await quickPickLocation(subscription);
+
+    const registry = await client.registries.beginCreate(resourceGroup.name, registryName, {
+        'sku': { 'name': sku },
+        'location': location
+    });
+    vscode.window.showInformationMessage(registry.name + ' has been created succesfully!');
+    dockerExplorerProvider.refreshRegistries();
+    return registry;
+}
+
+/** Acquires a new registry name from a user, validating that the name is unique */
+async function acquireRegistryName(client: ContainerRegistryManagementClient): Promise<string> {
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: false,
+        prompt: 'Enter the new registry name? '
+    };
+    let registryName: string = await ext.ui.showInputBox(opt);
+
+    let registryStatus: RegistryNameStatus = await client.registries.checkNameAvailability({ 'name': registryName });
+    while (!registryStatus.nameAvailable) {
+        opt = {
+            ignoreFocusOut: false,
+            prompt: `The registry name '${registryName}' is unavailable. Try again: `
```

Comment:
prompt: `The registry name '${registryName}' is unavailable. Try again.`

(Most prompts should not have periods, but since this contains multiple sentences, they're appropriate here.)